### PR TITLE
Better Canvas search to prevent metadata beeing searched in fulltext | Code formatted

### PR DESCRIPTION
**Pull request by CobriMediaJulien**
For reference:
https://github.com/TriliumNext/Notes/pull/720

@eliandoran
Could you pls check the code if the quality regarding variable names is okay? I just hacked it together. It kinda works well enough. i don´t have anymore time this week.

Now it searches only the text or topic of the notes.

### CODE REVIEW

File: src/services/search/expressions/note_content_fulltext.ts [hunk 1]
```diff
@@ -133,6 +133,26 @@ class NoteContentFulltextExp extends Expression {
 
             content = content.replace(/&nbsp;/g, ' ');
         }
+        else if (type === 'mindMap' && mime === 'application/json') {
+            let mindMapcontent = JSON.parse (content);
+            let topic = mindMapcontent.nodedata.topic;
+            content = utils.normalize(topic.toString());
```
Comment (eliandoran) [inline src/services/search/expressions/note_content_fulltext.ts:139]: This doesn't seem to go through all the nodes in order to search for content.
I added a text node randomly inside the tree and I couldn't find it.
Having the text in a canvas note instead appears to work fine.

**Review by eliandoran — CHANGES_REQUESTED** [review]

File: src/services/search/expressions/note_content_fulltext.ts (hunk 1 above)
Comment (CobriMediaJulien) [reply src/services/search/expressions/note_content_fulltext.ts:139]: Do you have further information?

I tried the same and it does work in my case. Random Text note inside my tree:

image

Comment (eliandoran) [reply src/services/search/expressions/note_content_fulltext.ts:139]: @CobriMediaJulien , for the mind map you would need to add more nodes. From what I see in the screenshot it's only a single node ("now a mindmaptest"). Try pressing enter a few times to create a few subnodes and look at that subnode instead.

I believe you need to iterate recursively through all the nodes, similarly to how you do for canvas note.

Comment (CobriMediaJulien) [reply src/services/search/expressions/note_content_fulltext.ts:139]: Could reproduce. I just search for topic value and not children. I will fix it this weekend.

Comment (CobriMediaJulien) [reply src/services/search/expressions/note_content_fulltext.ts]: Fixed and commited to this PR

Comment (eliandoran) [reply src/services/search/expressions/note_content_fulltext.ts]: @CobriMediaJulien , don't forget to remove the draft status from the PR once the required changes were done.

Comment (CobriMediaJulien) [reply src/services/search/expressions/note_content_fulltext.ts]: this PR provides also a fix for:

https://github.com/TriliumNext/Notes/issues/772

**Pull request closed without merging**